agent/structs/connect_ca.go

```diff
@@ -112,6 +112,53 @@ type CARoot struct {
 // CARoots is a list of CARoot structures.
 type CARoots []*CARoot
 
+// CAIntermediate represents an intermediate CA certificate that is trusted and used
+// to sign leaf certificates.
+type CAIntermediate struct {
```

Comment:
I'm not super sure about this. The original `CARoot` structure is perhaps confusingly named but it was designed to encapsulate all that was needed about a CA configuration (i.e. a _trust_ root) regardless of if it used a self-signed root or an Intermediate. The fact that this is (almost?) copy paste of that struct is a bit of a red flag.

Is there actually difference in functionality needed from `CARoot` for intermediate case?

My worry is that this superficially cleans up the semantics in one or two places but doesn't change the underlying assumption that a `CARoot` may also hold information about a CA that uses an intermediate cert which seems even more of a mess to me.

I need to re-read a lot of this to build enough context to know why it's needed or what the real clarification is though. @tradel can you help me recall why this was added?

Reply:
I'm not sure this is needed either - it's only used in one spot and if we only need the ID of the cert we can use `connect.CalculateCertFingerprint`.

Reply:
See my comment below, but the original justification was that in many cases, certs are issued by the intermediate CA, but when we check for root rotation, we only check leaf certs against the root CA.

Reply:
I just assumed that we might need more of the intermediate's details at some point, so rather than just return a string containing the signing key, I created a new struct and a `parseIntermediate` function. But you're right that it isn't necessary right now.

Reply:
@kyhavlov IMHO we should use `connect.HexString(cert.AuthorityKeyId)` to be consistent with what we're doing elsewhere (e.g. [here](https://github.com/hashicorp/consul/blob/d4443374cd872ebbef9b19265418c447233eed2f/agent/cache-types/connect_ca_leaf.go#L599)).